### ISSUE TITLE
Retract a module and swipl.call fail help

### ISSUE THREAD

**Issue by FerrariAndrea** [rla/node-swipl#8]
Hi, I have two questions,

Ther's a way to retract all caluse of a module?
Example: 

```
swipl.call('assert(mymodule:test(1))');
swipl.call('assert(mymodule:test(2))');
swipl.call('assert(mymodule:test(3))');
```
now i need a call like this:

`swipl.call('retractall(mymodule:_)');`

**Exist a way for retract all clause related to `mymodule`?**

And I noticed that, if i do a swipl.call('someclause') and the 'someclause' fail or not exits i receive an error like this:
```
[PROLOG SYSTEM ERROR:  Thread 1 (main) at Fri Nov 01 11:15:49 2019
        PL_close_foreign_frame(): illegal frame: 93

PROLOG STACK:
]
```
if i use `swipl.Query('someclause')` all be good, i receive `false`
EDIT:
if i close the Query i recive the same error of a call, example:

without the close:
  ```
 swipl.call('assert(thisIsATest(45))');
       const query = new swipl.Query('thisIsATest(5)');
        let ret = null;
        while (ret = query.next()) {
            console.log(`Variable X value is: ${ret.X}`);
        }        
     return(ret); 
```
this return false but then i can't open a new Query.

with the close:

  ```
 swipl.call('assert(thisIsATest(45))');
       const query = new swipl.Query('thisIsATest(5)');
        let ret = null;
        while (ret = query.next()) {
            console.log(`Variable X value is: ${ret.X}`);
        }        
        query .close();
     return(ret); 
```
i receive: 

```
[PROLOG SYSTEM ERROR:  Thread 1 (main) at Fri Nov 01 11:50:28 2019
        PL_close_foreign_frame(): illegal frame: 86

PROLOG STACK:
]

[pid=14216] Action?
```

**It's normal? any suggestion?**

**Comment by FerrariAndrea** [comment]
For the query.close() error, i noticed that: if query produce "fase" as result than you must not close the query.

For example:

 ```
       swipl.call('assert(thisIsATest(45))'); 
        var query = new swipl.Query('thisIsATest(5).');
        let ret = true;
        var ris ;
        while (ret!==false) {
            ret = query.next();
            console.log(`Variable X value is: ${ret}`);
            ris = ret;
        }  
        if(ris!==false){  //only if you have an result
            query.close();
        }  
```

**Comment by rla** [comment]
@FerrariAndrea, I'm hoping to work on this on the next week and figure out why the error happens with the `close` method.

For clearing a module, you might want to use something like this:
https://github.com/Blog-Core/blog-core/blob/e1423f4fe3f302ceafe4b2a8142f126b7b07a727/prolog/bc/bc_analytics_db.pl#L337

**Comment by rla** [comment]
@FerrariAndrea, the foreign frame error was caused by double-closing the foreign frame. I updated the code to avoid it and published fix with version 1.0.5 on npm. Please test.

**Comment by FerrariAndrea** [comment]
@rla, i confirm that now it's work correctly.

```
function test(){
   swipl.call('assert(thisIsATest(45))');
        const query = new swipl.Query('thisIsATest(5)');
        var ret = null;
        while (ret = query.next()) {
            console.log(`Variable X value is: ${ret.X}`);
        }        
         query .close();
     return(ret); 
}
```
this code return `false`, without error.

Thanks for your job.
